Assignment4/JavierSabin-190299/task08.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+"""Task08.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/18P0xTS31q1P7nR4efg0Yy3QrClOxqMAe
+
+**Task 08: Completing missing data**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2020-2021/master/Assignment4/"
+
+from rdflib import Graph, Namespace, Literal, URIRef
+g1 = Graph()
+g2 = Graph()
+g1.parse(github_storage+"resources/data01.rdf", format="xml")
+g2.parse(github_storage+"resources/data02.rdf", format="xml")
+
+"""Tarea: lista todos los elementos de la clase Person en el primer grafo (data01.rdf) y completa los campos (given name, family name y email) que puedan faltar con los datos del segundo grafo (data02.rdf). Puedes usar consultas SPARQL o iterar el grafo, o ambas cosas."""
+
+from rdflib.namespace import RDF
+from rdflib.plugins.sparql import prepareQuery
+
+print("Incomplete Graph")
+for s, p, o in g1:
+  print(s,p,o)
+
+data = Namespace("http://data.org#")
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+query1 = prepareQuery('''
+  SELECT 
+    ?Subject
+  WHERE { 
+    ?Subject rdf:type ns:Person
+  }
+  ''',
+  initNs = { "ns": data, "rdf": RDF}
+  )
+for r in g1.query(query1):
+  print(r.Subject)
+
+query2 = prepareQuery('''
+  SELECT 
+    ?Subject ?Predicate ?Object
+  WHERE {
+    ?Subject ?Predicate ?Object
+    FILTER(?Predicate=vcard:FN || ?Predicate=vcard:Given || ?Predicate=vcard:EMAIL)
+  }
+  ''',
+  initNs = { "vcard": vcard}
+  )
+
+for r in g2.query(query2):
+  print(r)
+  g1.add(r)
```

Comment:
This is not correct. You are adding all the subjects of g2 into g1. The question is asking to complete the elements of g1 from those that have more information in g2.

Reply:
I just uploaded the corrected version